src/main/java/org/takes/facets/fork/RsFork.java

```diff
@@ -62,17 +62,10 @@ public RsFork(final Request req, final Fork... list) {
      */
     public RsFork(final Request req, final Iterable<Fork> list) {
         super(
-            new Response() {
-                @Override
-                public Iterable<String> head() throws IOException {
-                    return RsFork.pick(req, list).head();
-                }
-
-                @Override
-                public InputStream body() throws IOException {
-                    return RsFork.pick(req, list).body();
-                }
-            }
+            new ResponseOf(
+                () -> RsFork.pick(req, list).head(),
```

Comment:
@fanifieiev use the seconday constructor instead.

Reply:
@fabriciofx Can you clarify here?
Do you mean creating new constructor that would accept `Response`?
Like this:
`public ResponseOf(final Response response) {
        this(
            response::head,
            response::body
        );
  }`

Reply:
@fabriciofx If you meant about the secondary constructor that accepts `head` and `body` and use the following:
![image](https://user-images.githubusercontent.com/3167940/68986904-286e0380-07f2-11ea-89b8-c17f283e8ec1.png)
then it won't work.
I do nto want the code to be executed inside the constructor, but rather pass in `Scalar` so that the code is executed when needed(lazy execution).
The other problem as can be seen on the screenshot, the code is not compiled as it reqyures exception throwing or catching in constructor.

Reply:
@fanifieiev I'm talking about it:
```
public final class ResponseOf implements Response {

    /**
     * Original head scalar.
     */
    private final Scalar<Iterable<String>> shead;

    /**
     * Original body scalar.
     */
    private final Scalar<InputStream> sbody;

    /**
     * Ctor.
     * @param head Iterable head value
     * @param body InputStream body value
     */
    public ResponseOf(
        final Iterable<String> head,
        final InputStream body
    ) {
        this(
            () -> head,
            () -> body
        );
    }

    /**
     * Ctor.
     * @param head Scalar to provide head value
     * @param body Scalar to provide body value
     */
    public ResponseOf(
       final Scalar<Iterable<String>> head,
        final Scalar<InputStream> body
    ) {
        this.shead = head;
        this.sbody = body;
    }
...
```
and you use it in some places like bellow:
```
new ResponseOf(
     new RsEmpty().head(),
     new RsVelocity.render(folder, template, params.get()
);
```
Of couse you can't use it in all places (e.g. beacause it can throws an `Exception`). So, in these places you can use the primary constructor...